client/src/services/data.js

```diff
@@ -0,0 +1,52 @@
+import {logDebug} from './log';
+
+const states = {};
```

Comment:
This doesn't feel like a "React-y" way to do it ... I would still use the useState hook and the setState function to update this and use a custom hook to manage this

Reply:
No, this was an attempt to work around duplicate fetch call outside of react.  Issue I'm having was due to React 18 in dev mode.  Did some reading on it to understand what's going and refactored all the code to useState & useEffect & properly assign dependencies so react can render each component correctly base on those dependencies.